src/plugins/intel_cpu/src/transformations/cpu_opset/convert_to_cpu_specific_opset.hpp

```diff
@@ -36,22 +38,24 @@ inline void ConvertToCPUSpecificOpset(std::shared_ptr<ov::Model>& model, const C
     CPU_REGISTER_PASS_COMMON(manager, ConvertMatMulToFC);
     CPU_REGISTER_PASS_COMMON(manager, FullyConnectedBiasFusion);
 
-    std::vector<ov::element::Type> supported_activation_types{
+    std::vector<ov::element::Type> supported_activation_types;
+    std::vector<ov::element::Type> supported_compressed_weights_types;
+
+    bool useMatmulPrim = false;
+    CPU_DEBUG_CAP_ENABLE(useMatmulPrim = getEnvBool("OV_CPU_ENABLE_DNNL_MAMTUL_FOR_FC");)
+
+    if (useMatmulPrim) {
+        supported_activation_types = {Type_t::f32, Type_t::f16};
+        supported_compressed_weights_types = {Type_t::u8, Type_t::i8};
+    } else {
         // @todo enable for bf16 as well
         // after EnforceInferencePrecision is replaced with ConvertPrecision
-        ov::element::f32,
-    };
+        supported_activation_types = {Type_t::f32};
+        supported_compressed_weights_types =
+            {Type_t::u8, Type_t::i8, Type_t::u4, Type_t::i4, Type_t::nf4, Type_t::f4e2m1};
+    }
 
-    std::vector<ov::element::Type> supported_compressed_weights_types{
-        ov::element::u8,
-        ov::element::i8,
-        ov::element::u4,
-        ov::element::i4,
-        ov::element::nf4,
-        ov::element::f4e2m1,
-    };
-
-    CPU_REGISTER_PASS_X64(
+    CPU_REGISTER_PASS_COMMON(
```

Comment:
I can put CPU_REGISTER_PASS_COMMON only for transformation which callback is responsible for arch related checks.
Since now I added getSupportedCompressedWeightsTypes  API, decompression markup can also be stated as common.

Reply:
Got it, thanks